pkg/eventmonitor/eventmonitor.go

```diff
@@ -139,16 +139,18 @@ func (m *EventMonitor) Start() error {
 		return err
 	}
 
-	if err := m.Probe.Start(); err != nil {
-		return err
-	}
-
 	// start event consumers
 	for _, em := range m.eventConsumers {
 		if err := em.Start(); err != nil {
 			log.Errorf("unable to start %s event consumer: %v", em.ID(), err)
 		}
 	}
+	// Apply rules to the snapshotted data
+	m.Probe.PlaySnapshot()
```

Comment:
If we can I would call the PlaySnapshot function directly from the Start function